Engine/ProfilerCore.cs

```diff
@@ -0,0 +1,104 @@
+﻿using HarmonyLib;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Reflection;
+using System.Diagnostics;
+using System.Runtime.CompilerServices;
+
+namespace Engine
+{
+    public static class ProfilerCore
+    {
+        private static readonly Harmony _harmony = new("Profiler");
+        private static readonly List<MethodInfo> _methodsToPatch = new();
+        private static readonly List<ProfilingResult> _results = new();
```

Comment:
1) please remove all underscore prefixes to follow the project's code style
2) some tree-like structure is needed to keep track of methods being profiled inside another methods being profiled